lib/Basic/UUID.cpp

```diff
@@ -15,40 +15,126 @@
 //
 //===----------------------------------------------------------------------===//
 
-#include <uuid/uuid.h>
 #include "swift/Basic/UUID.h"
 
+// WIN32 doesn't natively support <uuid/uuid.h>. Instead, we use Win32 APIs
+// class
+#if defined(_WIN32)
+#define WIN32_LEAN_AND_MEAN
+#include <rpc.h>
+#include <string>
+
+// Windows has a different layout of UUID than libuuid. However, they are the same
+// size, as mandated by the UUID specification (RFC 4122). This means that we can
+// convert the memory of a 128 bit unsigned char array to a ::UUID.
+// We also can't expose <rpc.h> in any headers, as it imports <windows.h> which
+// defines macros that conflict with many definitions (e.g. ERROR, std::max)
+union Conversion {
```

Comment:
Either ::UUID is a [POD type](http://en.cppreference.com/w/cpp/concept/PODType), in which case you can just memcpy into one directly, or it's not, in which case the union isn't safe either.